internal/repository/repository.go

```diff
@@ -252,12 +322,51 @@ func (r *Repository) LookupBlobSize(id restic.ID, tpe restic.BlobType) (uint, bo
 	return r.idx.LookupSize(restic.BlobHandle{ID: id, Type: tpe})
 }
 
+func (r *Repository) getZstdEncoder() *zstd.Encoder {
+	r.allocEnc.Do(func() {
+		level := zstd.SpeedDefault
+		if r.opts.Compression == CompressionMax {
+			level = zstd.SpeedBestCompression
+		}
```

Comment:
From my perspective, even though named constants are nice - I'd much appreciate the ability to specify the compression level numerically just as if I were using the gzip, xz, bzip2, etc.

Reply:
restic processes multiple blocks in parallel. so a slight performance increase by a lower compression level won't make that much of a difference in total. Especially, as the compression is not the only processing step.

What is the benefit of providing tons of different compression level?

Reply:
For an actual use case - Borg backup exposes zstd + level as options and that allows me to fine tune the overall cpu/runtime/compression of my very large DR backup jobs (backup of backup server basically).  Would it be the end of the world for me if I can't do the same tuning in restic?  Not really.  However, I do currently find actual value in having that level of control.

Reply:
The go implementation of zstandard doesn’t use numeric levels but only descriptive levels: https://github.com/klauspost/compress/blob/c06ba5f93c87a4bb844648f972ac3db430cb4da2/zstd/encoder_options.go#L46

Anyway, two levels (normal and max compression) are more than enough imho, instead I’d spend some time trying to figure out the best parameters for the normal level, since 
breaking files into separate chunks of predictable average size that are then compressed independently, like restic does, is not the most common use case.

Reply:
Fair enough - sounds like a lot of extra work for minimal utility.